index.js

```diff
@@ -158,6 +158,36 @@ Object.defineProperty(File.prototype, 'relative', {
   }
 });
 
+Object.defineProperty(File.prototype, 'dirname', {
+  get: function() {
+    if (!this.path) throw new Error('No path specified! Can not get dirname.');
+    return path.dirname(this.path);
+  },
+  set: function() {
+    throw new Error('File.dirname is generated from the base and path attributes. Do not modify it.');
+  }
+});
+
+Object.defineProperty(File.prototype, 'basename', {
+  get: function() {
+    if (!this.path) throw new Error('No path specified! Can not get basename.');
+    return path.basename(this.path);
+  },
+  set: function() {
+    throw new Error('File.basename is generated from the base and path attributes. Do not modify it.');
+  }
+});
+
+Object.defineProperty(File.prototype, 'extname', {
+  get: function() {
+    if (!this.path) throw new Error('No path specified! Can not get extname.');
+    return path.extname(this.path);
+  },
+  set: function() {
+    throw new Error('File.extname is generated from the base and path attributes. Do not modify it.');
```

Comment:
i do agree that if we have a setter for this it should use the replace-ext module on path